src/StackExchange.Redis/ExceptionFactory.cs

```diff
@@ -124,6 +124,11 @@ internal static string GetInnerMostExceptionMessage(Exception? e)
                 // This can happen in cloud environments often, where user disables abort and has the wrong config
                 initialMessage = $"Connection to Redis never succeeded (attempts: {attempts} - check your config), unable to service operation: ";
             }
+            else if (message is not null && message.IsPrimaryOnly())
```

Comment:
I might be missing some context here, isn't this a bit inferential?

Reply:
I'm not sure I follow - how so? If we're passed a message and we know it's demanding a primary, we're simply adding that info to the message.

Reply:
yeah, but this could happen in any number of failure scenarios right?

This is kind of implying that you might have been able to connect to a replica, but connecting to a primary was the issue.

You could theoretically see this in any topography, couldn't you?

Reply:
Correct, the message isn't meant to imply there is a replica, alternative phrasing suggestion?

Reply:
Maybe add to the conditional that you need to be on a topography with replication? It looks like the multiplexers configuration is being interrogated above, so it doesn't seems like that would be totally out-of-band behavior here. And then you can use something along the lines of `"No writeable connection active/available to service this operation: "`?

Reply:
I tried that at first but felt phrasing was too little to indicate what's happening. The check we can do though - I can add a conditional to being connected to anything. From that, we can infer we only have replica connections. Pushed up a tweak here, and indeed the async test _won't_ get this message now, because it's only connecting to a primary and it's not available, so gets the old message.

Reply:
Sounds good 👍